subsys/fs/nvs/nvs.c

```diff
@@ -473,7 +485,7 @@ static int _nvs_update_free_space(struct nvs_fs *fs)
 
 	int rc;
 	struct nvs_ate step_ate, wlk_ate;
-	u32_t step_addr, wlk_addr;
+	u32_t step_addr, end_addr, wlk_addr;
```

Comment:
this is what i noticed - pleas let me know if i'm wrong

Reply:
_nvs_prev_ate does not modify fs.ate_wra. I tought it was more clear this way. However since I am comparing wlk_addr with fs.ate_wra I should probably also compare step_addr with this.